sklearn/datasets/openml.py

```diff
@@ -0,0 +1,200 @@
+                                  + "/data_version/{}".format(version))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+        else:
+            raise error
+
+    if not data_found and version != "active":
+        # might have been deactivated. will warn later
+        data_found = True
+        try:
+            json_string = urlopen(_SEARCH_NAME.format(name) +
+                                  "/data_version/{}/status/deactivated".format(
+                                      version))
+        except HTTPError as error:
+            if error.code == 412:
+                data_found = False
+            else:
+                raise error
+
+    if not data_found:
+        # not in except for nicer traceback
+        if version == "active":
+            raise ValueError("No active dataset {} found.".format(name))
+        raise ValueError("Dataset {} with version {}"
+                         " not found.".format(name, version))
+
+    json_data = json.load(json_string)
+    return json_data['data']['dataset'][0]
+
+
+def _get_data_description_by_id(data_id):
+    data_found = True
+    try:
+        json_string = urlopen(_DATA_INFO.format(data_id))
+    except HTTPError as error:
+        if error.code == 412:
+            data_found = False
+    if not data_found:
+        # not in except for nicer traceback
+        raise ValueError("Dataset with id {} "
+                         "not found.".format(data_id))
+    json_data = json.load(json_string)
+    return json_data['data_set_description']
+
+
+def _download_data(url):
+    response = urlopen(url)
+    if sys.version_info[0] == 2:
+        # Python2.7 numpy can't handle unicode?
+        arff = loadarff(StringIO(response.read()))
+    else:
+        arff = loadarff(StringIO(response.read().decode('utf-8')))
+
+    response.close()
+    return arff
+
+
+def fetch_openml(name_or_id=None, version='active', data_home=None,
+                 target_column='default-target', memory=True):
+    """Fetch dataset from openml by name or dataset id.
+
+    Datasets are uniquely identified by either an integer ID or by a
+    combination of name and version (i.e. there might be multiple
+    versions of the 'iris' dataset).
+
+    Parameters
+    ----------
+    name_or_id : string or integer
+        Identifier of the dataset. If integer, assumed to be the id of the
+        dataset on OpenML, if string, assumed to be the name of the dataset.
+
+    version : integer or 'active', default='active'
+        Version of the dataset. Only used if ``name_or_id`` is a string.
+        If 'active' the oldest version that's still active is used.
+
+    data_home : string or None, default None
+        Specify another download and cache folder for the data sets. By default
+        all scikit-learn data is stored in '~/scikit_learn_data' subfolders.
+
+    target_column : string or None, default 'default-target'
```

Comment:
do we want to handle multiple targets?

Reply:
OpenML contains several multi-target datasets